webui/react/src/pages/TaskList.settings.ts

```diff
@@ -0,0 +1,81 @@
+import { MINIMUM_PAGE_SIZE } from 'components/Table';
+import { BaseType, SettingsConfig } from 'hooks/useSettings';
+import { CommandState, CommandType } from 'types';
+
+export interface Settings {
+  search: string;
+  sortDesc: boolean;
+  sortKey: 'id' | 'name' | 'resourcePool' | 'startTime' | 'state' | 'type' | 'username';
+  state: CommandState[] | undefined;
+  tableLimit: number;
+  tableOffset: number;
+  type: CommandType[] | undefined;
+  user: string[] | undefined;
+}
```

Comment:
I started off using `io-ts` + `io.TypeOf` to create the Typescript definitions, but there are too many walls I've hit. The biggest problem is that as definitions became more complex, the generated Typescript became `any`. Anytime, `io.Mixed` is used `any` gets generated, which renders the Typescript useless.